src/DynamoCore/Configuration/PathManager.cs

```diff
@@ -131,6 +131,29 @@ internal static string BuiltinPackagesDirectory
             }
         }
 
+        private static char[] invalidCharacters = null;
+        /// <summary>
+        /// List of the Windows invalid characters
+        /// </summary>
+        internal static char[] InvalidCharacters
+        {
+            get
+            {
+                if (invalidCharacters == null)
+                {
+                    invalidCharacters = Search.SearchDictionary<object>.SpecialAndInvalidCharacters();
```

Comment:
Looking at this PR again, I still would not prefer this way. The original getter of invalid chars lives inside of search dictionary which seems strange. Can you move this to https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoUtilities/PathHelper.cs and have `ContainsSpecialCharacters` refer to this property?